lua/tabnine/auto_commands.lua

```diff
@@ -38,6 +39,13 @@ function M.setup()
 			end
 		end,
 	})
+
+	api.nvim_create_autocmd("VimEnter", {
+		pattern = "*",
+		callback = function()
+			chat_setup.setup()
```

Comment:
In `/lua/tabnine/auto_commands.lua`, you call `chat_setup.setup()` on `VimEnter`, but this is a bad idea, as you can't be certain `VimEnter` will *ever* fire. In my case, using `Lazy.nvim`, since I lazy load tabnine-nvim, it never gets fired (tabnine gets loaded **after** VimEnter). A better way about this would be to call `chat_setup.setup()` in `/lua/tabnine.lua` during the main `setup()`

Reply:
Here's a patch that does this:

```patch
From 03b67fa0fc0a20252336c9e7e53cde9c796812e8 Mon Sep 17 00:00:00 2001
From: aarondill <aaronsacks2006@gmail.com>
Date: Fri, 7 Jul 2023 02:19:34 -0500
Subject: [PATCH] setup tabnine chat in tabnine.lua

---
 lua/tabnine.lua               | 3 +++
 lua/tabnine/auto_commands.lua | 8 --------
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/lua/tabnine.lua b/lua/tabnine.lua
index bfb83d9..f835546 100644
--- a/lua/tabnine.lua
+++ b/lua/tabnine.lua
@@ -5,6 +5,7 @@ local auto_commands = require("tabnine.auto_commands")
 local user_commands = require("tabnine.user_commands")
 local status = require("tabnine.status")
 local keymaps = require("tabnine.keymaps")
+local chat_setup = require("tabnine.chat.setup")
 
 local M = {}
 
@@ -35,6 +36,8 @@ function M.setup(o)
 	auto_commands.setup()
 
 	status.setup()
+
+	chat_setup.setup()
 end
 
 return M
diff --git a/lua/tabnine/auto_commands.lua b/lua/tabnine/auto_commands.lua
index d58735b..9bfa0b2 100644
--- a/lua/tabnine/auto_commands.lua
+++ b/lua/tabnine/auto_commands.lua
@@ -3,7 +3,6 @@ local consts = require("tabnine.consts")
 local state = require("tabnine.state")
 local completion = require("tabnine.completion")
 local config = require("tabnine.config")
-local chat_setup = require("tabnine.chat.setup")
 
 local M = {}
 
@@ -39,13 +38,6 @@ function M.setup()
 			end
 		end,
 	})
-
-	api.nvim_create_autocmd("VimEnter", {
-		pattern = "*",
-		callback = function()
-			chat_setup.setup()
-		end,
-	})
 end
 
 return M
-- 
2.34.1
```

Reply:
done